omod/src/test/java/org/openmrs/module/htmlformentryui/fragment/controller/htmlform/EnterHtmlFormFragmentControllerComponentTest.java

```diff
@@ -112,8 +112,9 @@
 
     UiSessionContext sessionContext;
 
-    public static final String FORM_DEFINITION = "<htmlform formUuid=\"form-uuid\" formName=\"Form Name\" formVersion=\"1.0\">Weight:<obs id=\"weight\" conceptId=\"5089\"/> <encounterDate showTime=\"true\"/> <encounterLocation/> <encounterProvider/></htmlform>";
-    public static final String FEATURE_TOGGLE_TEST_FORM_DEFINITION = "<htmlform formUuid=\"form-uuid\" formName=\"Form Name\" formVersion=\"1.0\"> <includeIf velocityTest=\"$featureToggles.isFeatureEnabled('someFeatureToggle')\"> featureToggledOn </includeIf>"
+    public static final String FORM_DEFINITION = "<htmlform formUuid=\"form-uuid\" formName=\"Form Name\" formVersion=\"1.0\" formEncounterType=\"61ae96f4-6afe-4351-b6f8-cd4fc383cce1\">Weight:<obs id=\"weight\" conceptId=\"5089\"/> <encounterDate showTime=\"true\"/> <encounterLocation/> <encounterProvider/></htmlform>";
+    
+    public static final String FEATURE_TOGGLE_TEST_FORM_DEFINITION = "<htmlform formUuid=\"form-uuid\" formName=\"Form Name\" formVersion=\"1.0\" formEncounterType=\"61ae96f4-6afe-4351-b6f8-cd4fc383cce1\"> <includeIf velocityTest=\"$featureToggles.isFeatureEnabled('someFeatureToggle')\"> featureToggledOn </includeIf>"
```

Comment:
@mogoodrich @mseaton this test breaks on a missing encounter ID Core validation if those two test resources don't specify a `formEncounterType` attribute.

If they do specify one then [this line](https://github.com/openmrs/openmrs-module-htmlformentry/blob/666e7ab77456176aae8753c9540a55c8600b036d/api/src/main/java/org/openmrs/module/htmlformentry/FormEntrySession.java#L627) in `FormEntrySession` is reached and all is fine.

Any idea?

Reply:
No - maybe the previous version of HFE (or core) also defined a form with uuid of "form-uuid", which already had an encounter type associated with it, and this was just modifying it?

Reply:
Seems like a reasonable supposition @mseaton